mvpa2/clfs/transerror.py

```diff
@@ -969,6 +969,10 @@ def plot(self, labels=None, numbers=False, origin='upper',
                     v = None
                     if l is not None: v = labels_map_full[l]
                     labels_plot += [v]
+            elif (len(labels) != labels_order_filtered_set) and \
+                    all(isinstance(x, str) for x in labels_order_filtered_set):
```

Comment:
type shouldn't matter really - I might be having numeric labels
also, why not just to use sets operation to check exactly that `labels_order_filtered_set` includes all we would like to plot like
```suggestion
            elif not (set(labels) - labels_order_filtered_set):
```
? because `len` is too weak of a check -- I could have completely different labels from what is provided to labels_order and condition would still pass, but then you would end up with no `labels_plot`